tests/Properties.hs

```diff
@@ -781,6 +781,39 @@ prop_index_C (String8 xs) =
     forAll indices $ \i -> (xs !! i) == C.pack xs `C.index` (fromIntegral i)
   where indices = choose (0, length xs -1)
 
+-- | Test 'indexMaybe' for Lazy and Strict 'ByteString's.
```

Comment:
How about simplifying these tests a bit:

Generate bytestrings and arbitrary indices, and test that: if the index is in the bounds, we get a `Just`, otherwise a `Nothing`.

That should also help with test performance.

Also, the strict versions of `indexMaybe` are so trivial that IMHO they are hardly worth testing, but I don't care too much.